### PR TITLE
TandemFoil: weight-decay=1e-2 regularization (was default)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The current TandemFoil baseline uses default weight decay (likely 0 or very small). Adding weight-decay=1e-2 may act as a regularizer to reduce overfitting, especially given TandemFoil's smaller dataset size. This is a low-cost experiment with potentially high signal on whether the model is capacity-limited vs. regularization-limited.

## Instructions
Run the following command exactly:

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 3e-4 \
  --cosine-t-max 10 \
  --weight-decay 1e-2 \
  --no-use-ema \
  --model-slices 64 \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb-name tandem-5L256d-wd1e2
```

Changes vs baseline: `--weight-decay 1e-2` added.

## Baseline
- TandemFoil val: 52.81 / test: 55.25
- Baseline W&B run: `l5kggnbg`
- PR: #2595, 67 epochs, 180-min wall-clock
- Reproduce: `cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --model-layers 5 --model-hidden-dim 256 --model-heads 4 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2713 because ADVISOR: Closing — 5L/256d+WD at lr=3e-4 superseded by lr=2e-4..

**Comment by tcapelle** [comment]
STUDENT levi:

## Results (completed before PR closure)

### TandemFoil 5L/256d wd=1e-2 (W&B run dmo3eq04)
- **val_primary/surface_pressure_mae:** 56.88 (final, epoch 66) — **best: 48.95 at epoch 63** (-7.3% vs baseline 52.81)
- **test_primary/surface_pressure_mae:** 54.65 (final)
- **Per-split test MAE:** single_in_dist=58.49, geom_camber_rc=59.35, geom_camber_cruise=52.20, re_rand=48.57
- **Epochs:** 66 (183 min)
- **W&B run:** dmo3eq04
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 --weight-decay 1e-2 --no-use-ema --model-slices 64 --model-layers 5 --model-hidden-dim 256 --model-heads 4 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999 --agent levi --wandb-name "levi/tandem-5L256d-wd1e2"`

### What happened
- **Weight decay=1e-2 helps**: best val=48.95 is a 7.3% improvement over baseline 52.81.
- Steady descent through training — model still improving at cutoff (epoch 63 was best).
- Some LR cycling instability (e.g., epoch 60: val=61.96 vs epoch 63: val=48.95), but less severe than Lion on AirfRANS.
- The re_rand split (48.57 test) benefits most — regularization helps generalization to random Reynolds numbers.

### Suggested follow-ups
- Combine with the lr=2e-4 that supersedes this experiment
- Try wd=5e-3 as a middle ground
- Model checkpointing at best val would capture 48.95 vs final 56.88
